tests/text_area/test_selection_bindings.py

```diff
@@ -13,34 +13,48 @@
 
 
 class TextAreaApp(App):
+    def __init__(self, read_only: bool = False):
+        super().__init__()
+        self.read_only = read_only
+
     def compose(self) -> ComposeResult:
-        text_area = TextArea(show_line_numbers=True)
-        text_area.load_text(TEXT)
-        yield text_area
+        yield TextArea(TEXT, show_line_numbers=True, read_only=self.read_only)
+
+
+@pytest.fixture(params=[True, False])
+async def read_only_mode(request):
+    """This parametrised fixture is injected into the app fixture.
+    It means every test that uses the app fixture will run twice - once
+    for read_only=True and once for read_only=False.
+    """
+    return request.param
+
 
+@pytest.fixture
+async def app(read_only_mode: bool):
+    text_area_app = TextAreaApp(read_only_mode)
+    yield text_area_app
```

Comment:
From the way I'm reading these two fixtures, it looks like you could just parametrise the second one and get rid of the first! 🚀

Reply:
Good spot!